File: scalang/src/main/scala/com/kapilkoju/autopos/accounting/api/DayBookEntryApi.scala
package com.kapilkoju.autopos.accounting.api

import com.kapilkoju.autopos.accounting.domain.DayBookEntry
import com.kapilkoju.autopos.accounting.repo.DayBookEntryRepo
import com.kapilkoju.autopos.kernel.api.BaseApi
import org.springframework.web.bind.annotation.{RequestMapping, RestController}

@RestController
@RequestMapping(Array("/api/day-book-entries"))
class DayBookEntryApi(private val dayBookEntryRepo: DayBookEntryRepo)
  extends BaseApi[DayBookEntry](dayBookEntryRepo, "dayBookEntry", "/api/day-book-entries/")

File: src/main/kotlin/com/kapilkoju/autopos/accounting/api/DayBookEntryApi.kt
package com.kapilkoju.autopos.accounting.api


import com.kapilkoju.autopos.accounting.domain.DayBookEntry
import com.kapilkoju.autopos.accounting.service.DayBookEntryRepo
import com.kapilkoju.autopos.kernel.api.BaseApi
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(DayBookEntryApi.baseUrl)
class DayBookEntryApi(dayBookEntryRepo: DayBookEntryRepo)
    : BaseApi<DayBookEntry>(dayBookEntryRepo, "dayBookEntry", baseUrl) {

    companion object {
        const val baseUrl = "/api/day-book-entries"
    }
}
